pkg/core/oracle_test.go

```diff
@@ -126,46 +143,78 @@ func TestOracle_InvalidWallet(t *testing.T) {
-	putOracleRequest(t, cs.Hash, bc, "https://get.maxallowed", nil, "handle", []byte{}, 100_000_000)
+	nativeOracleState := bc.GetContractState(nativeOracleH)
+	require.NotNil(t, nativeOracleState)
+	md := nativeOracleState.Manifest.ABI.GetMethod(manifest.MethodVerify, -1)
+	require.NotNil(t, md)
+	orc1.UpdateNativeContract(nativeOracleState.NEF.Script, slice.Copy(oracleRespScript), nativeOracleH, md.Offset)
+	orc2.UpdateNativeContract(nativeOracleState.NEF.Script, slice.Copy(oracleRespScript), nativeOracleH, md.Offset)
+
+	cs := contracts.GetOracleContractState(t, pathToInternalContracts, validator.ScriptHash(), 0)
+	rawManifest, err := json.Marshal(cs.Manifest)
+	require.NoError(t, err)
+	rawNef, err := cs.NEF.Bytes()
+	require.NoError(t, err)
+	tx := managementInvoker.PrepareInvoke(t, "deploy", rawNef, rawManifest)
+	e.AddNewBlock(t, tx)
+	e.CheckHalt(t, tx.Hash())
+	cInvoker := e.ValidatorInvoker(cs.Hash)
+
+	putOracleRequest(t, cInvoker, "https://get.1234", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.1234", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.timeout", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.notfound", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.forbidden", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://private.url", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.big", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.maxallowed", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.maxallowed", nil, "handle", []byte{}, 100_000_000)
 
 	flt := "$.Values[1]"
-	putOracleRequest(t, cs.Hash, bc, "https://get.filter", &flt, "handle", []byte{}, 10_000_000)
-	putOracleRequest(t, cs.Hash, bc, "https://get.filterinv", &flt, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.filter", &flt, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.filterinv", &flt, "handle", []byte{}, 10_000_000)
 
-	putOracleRequest(t, cs.Hash, bc, "https://get.invalidcontent", nil, "handle", []byte{}, 10_000_000)
+	putOracleRequest(t, cInvoker, "https://get.invalidcontent", nil, "handle", []byte{}, 10_000_000)
 
 	checkResp := func(t *testing.T, id uint64, resp *transaction.OracleResponse) *state.OracleRequest {
-		req, err := oracleCtr.GetRequestInternal(bc.dao, id)
-		require.NoError(t, err)
+		// Use a hack to get request from Oracle contract, because we can't use GetRequestInternal directly.
+		requestKey := make([]byte, 9)
+		requestKey[0] = 7 // prefixRequest from native Oracle contract
+		binary.BigEndian.PutUint64(requestKey[1:], id)
+		si := bc.GetStorageItem(nativeOracleID, requestKey)
+		require.NotNil(t, si)
+		req := new(state.OracleRequest)
+		require.NoError(t, stackitem.DeserializeConvertible(si, req))
```

Comment:
And here's one more dirty hack in order to avoid call to unexported Oracle's `GetRequestInternal`. Is it appropriate for us?

Reply:
Could you elaborate, why can't we use `GetRequestInternal`? It seems to be exported.

Reply:
The question is still relevant, can we keep this test in the form it's implemented now?

Reply:
But we don't have DAO outside of the blockchain.